### PR TITLE
STORM-1016: Generate trident bolt ids with sorted group names

### PR DESCRIPTION
**Pull request by victor-wong**
PR for [STORM-1016](https://issues.apache.org/jira/browse/STORM-1016)

### CODE REVIEW

**Comment by revans2** [comment]
@victor-wong these changes look good, but would it be cleaner to just use a TreeMap instead of a HashMap?

**Comment by revans2** [comment]
@victor-wong any update on this?

**Comment by victor-wong** [comment]
@revans2 sorry for replying so late, I just made some changes and it worked OK. Is that what you mean?

**Comment by revans2** [comment]
@victor-wong yes that is exactly what I was thinking, makes the code a lot smaller.  I am +1 on this change, but it looks like there is a code conflict.  Could you upmerge your changes?

And don't worry about it taking a while.

**Comment by victor-wong** [comment]
@revans2 changes upmerged. Is that right?

**Comment by revans2** [comment]
The build is complaining that storm-core/test/jvm/storm/trident/TestTridentTopology.java does not have an apache license header.

File: storm-core/src/jvm/storm/trident/graph/GraphGrouper.java [hunk 1]
```diff
@@ -34,8 +31,8 @@
     
     public GraphGrouper(DirectedGraph<Node, IndexedEdge> graph, Collection<Group> initialGroups) {
         this.graph = graph;
-        this.currGroups = new HashSet<>(initialGroups);
-        reindex();      
+        this.currGroups = new LinkedHashSet(initialGroups);
```
Comment (revans2) [inline storm-core/src/jvm/storm/trident/graph/GraphGrouper.java]: Could you put the `<>` characters back at the end of the LinkedHashSet?

**Comment by revans2** [comment]
Two very minor things, and I am +1

**Comment by victor-wong** [comment]
done :)

**Comment by revans2** [comment]
+1

**Pull request merged**